catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/CatalogFrameworkImpl.java

```diff
@@ -1082,6 +1285,43 @@ public QueryResponse query(QueryRequest queryRequest, FederationStrategy strateg
 
             validateFixQueryResponse(queryResponse, queryReq, overrideFanoutRename);
 
+            if (policy != null) {
+                HashMap<String, Set<String>> responsePolicyMap = new HashMap<>();
+                Map<String, Serializable> unmodifiableProperties = Collections.unmodifiableMap(
+                        queryReq.getProperties());
+                for (Result result : queryResponse.getResults()) {
+                    HashMap<String, Set<String>> itemPolicyMap = new HashMap<>();
+                    for (PolicyPlugin plugin : policy) {
```

Comment:
I see what you were talking about above now. No, I don't think it should be called in reverse order.